server/src/main/java/com/linecorp/centraldogma/server/internal/api/HttpApiUtil.java

```diff
@@ -190,7 +190,11 @@ private static HttpResponse newResponse0(RequestContext ctx, HttpStatus status,
 
         // TODO(minwoox) refine the error message
         try {
-            return HttpResponse.of(status, MediaType.JSON_UTF_8, Jackson.writeValueAsBytes(node));
+            if (status == HttpStatus.NO_CONTENT) {
+                return HttpResponse.of(HttpStatus.NO_CONTENT);
```

Comment:
Instead, how about adding a validation code in the beginning of this method? i.e.

```java
HttpResponse newResponse0(...) {
    checkArgument(!HttpStatusClass.INFORMATIONAL.contains(status.code()),
                  "...");
    switch (status.code()) {
        case /* NO_CONTENT */ 204:
        case /* RESET_CONTENT */ 205:
        case /* NOT_MODIFIED */ 304:
            throw new IllegalArgumentException(...);
    }
}
```

Reply:
Created a pull request for Armeria to simplify this:
https://github.com/line/armeria/pull/2058

Could you add `TODO` comment about this? i.e.

```java
// TODO(trustin): Use HttpStatus.isContentEmpty().
//                https://github.com/line/armeria/pull/2058
```

Reply:
> Instead, how about adding a validation code in the beginning of this method? i.e.
> 
> ```java
> HttpResponse newResponse0(...) {
>     checkArgument(!HttpStatusClass.INFORMATIONAL.contains(status.code()),
>                   "...");
>     switch (status.code()) {
>         case /* NO_CONTENT */ 204:
>         case /* RESET_CONTENT */ 205:
>         case /* NOT_MODIFIED */ 304:
>             throw new IllegalArgumentException(...);
>     }
> }
> ```

Sorry, I don't understand it...
I think If you throw an `IllegalArgumentException`, It will respond to `500 Internal Server Error`.
Or should return `HttpResponse.of(status)` in the switch-statement?
```java
    checkArgument(!HttpStatusClass.INFORMATIONAL.contains(status.code()),
                  "...");
    switch (status.code()) {
        case /* NO_CONTENT */ 204:
        case /* RESET_CONTENT */ 205:
        case /* NOT_MODIFIED */ 304:
            throw HttpResponse.of(status);
    }
```

Reply:
The intention of the suggested change is to make sure the caller of the method does not call it with such HttpStatuses, i.e. it would be a bug to call it with the HttpStatus whose response body will always be empty, because we do not want to let the caller waste its time preparing the response message.

That being said, could you make sure we do never call this method with such HttpStatuses?

Reply:
I fully understood. Thank you for the detailed explanation. 🙏 
Fixed in 9e869bc